{"cvrfdoc":{"document_title":"Red Hat Security Advisory: python38:3.8 security, bug fix, and enhancement update","document_distribution":"Copyright Â© 2021 Red Hat, Inc. All rights reserved.","document_references":{"reference":[{"description":"https://access.redhat.com/errata/RHSA-2020:4641","type":"Self","url":"https://access.redhat.com/errata/RHSA-2020:4641"},{"description":"https://access.redhat.com/security/updates/classification/#moderate","type":"External","url":"https://access.redhat.com/security/updates/classification/#moderate"},{"description":"https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.3_release_notes/","type":"External","url":"https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.3_release_notes/"}]},"aggregate_severity":"Moderate","document_tracking":{"initial_release_date":"2020-11-04T01:25:00Z","identification":{"id":"RHSA-2020:4641"},"revision_history":{"revision":{"date":"2020-11-04T01:25:00Z","number":"1","description":"Current version"}},"generator":{"date":"2021-08-23T04:54:00Z","engine":"Red Hat SDEngine 3.0.0"},"current_release_date":"2020-11-04T01:25:00Z","version":"1","status":"Final"},"product_tree":{"relationship":[{"relates_to_product_reference":"AppStream-8.3.0.GA","product_reference":"python38:3.8-8030020200818121840.4190259b","full_product_name":{"product_id":"AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b","cpe":"cpe:/a:redhat:enterprise_linux:8::appstream","product_name":"python38:3.8-8030020200818121840.4190259b as a component of Red Hat Enterprise Linux AppStream (v. 8)"},"relation_type":"Default Component Of"}],"branch":[{"name":"Red Hat Enterprise Linux","type":"Product Family","branch":[{"name":"Red Hat Enterprise Linux AppStream (v. 8)","full_product_name":{"product_id":"AppStream-8.3.0.GA","cpe":"cpe:/a:redhat:enterprise_linux:8::appstream","product_name":"Red Hat Enterprise Linux AppStream (v. 8)"},"type":"Product Name"}]},{"name":"python38:3.8-8030020200818121840.4190259b","full_product_name":{"product_id":"python38:3.8-8030020200818121840.4190259b","cpe":"cpe:/a:redhat:enterprise_linux:8::appstream","product_name":"python38-3.8-8030020200818121840.4190259b"},"type":"Product Version"}]},"document_publisher":{"issuing_authority":"Red Hat Product Security","contact_details":"secalert@redhat.com","type":"Vendor"},"vulnerability":[{"notes":{"note":"PyYAML: command execution through python/object/apply constructor in FullLoader"},"cve":"CVE-2019-20477","references":{"reference":[{"description":"CVE-2019-20477","url":"https://access.redhat.com/security/cve/CVE-2019-20477"},{"description":"bz#1806005: CVE-2019-20477 PyYAML: command execution through python/object/apply constructor in FullLoader","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1806005"}]},"release_date":"2019-11-18T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2020:4641"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-02-21T00:00:00Z","ordinal":"1"},{"notes":{"note":"python: infinite loop in the tarfile module via crafted TAR archive"},"cve":"CVE-2019-20907","references":{"reference":[{"description":"CVE-2019-20907","url":"https://access.redhat.com/security/cve/CVE-2019-20907"},{"description":"bz#1856481: CVE-2019-20907 python: infinite loop in the tarfile module via crafted TAR archive","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1856481"}]},"release_date":"2019-12-10T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2020:4641"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-07-13T00:00:00Z","ordinal":"2"},{"notes":{"note":"PyYAML: arbitrary command execution through python/object/new when FullLoader is used"},"cve":"CVE-2020-1747","references":{"reference":[{"description":"CVE-2020-1747","url":"https://access.redhat.com/security/cve/CVE-2020-1747"},{"description":"bz#1807367: CVE-2020-1747 PyYAML: arbitrary command execution through python/object/new when FullLoader is used","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1807367"}]},"release_date":"2020-03-02T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2020:4641"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"acknowledgments":{"acknowledgment":{"description":"This issue was discovered by Riccardo Schirone (Red Hat)."}},"discovery_date":"2020-02-26T00:00:00Z","ordinal":"3"},{"notes":{"note":"python: wrong backtracking in urllib.request.AbstractBasicAuthHandler allows for a ReDoS"},"cve":"CVE-2020-8492","references":{"reference":[{"description":"CVE-2020-8492","url":"https://access.redhat.com/security/cve/CVE-2020-8492"},{"description":"bz#1809065: CVE-2020-8492 python: wrong backtracking in urllib.request.AbstractBasicAuthHandler allows for a ReDoS","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1809065"}]},"release_date":"2020-03-02T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2020:4641"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-03-02T00:00:00Z","ordinal":"4"},{"notes":{"note":"python: DoS via inefficiency in IPv{4,6}Interface classes"},"cve":"CVE-2020-14422","references":{"reference":[{"description":"CVE-2020-14422","url":"https://access.redhat.com/security/cve/CVE-2020-14422"},{"description":"bz#1854926: CVE-2020-14422 python: DoS via inefficiency in IPv{4,6}Interface classes","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1854926"}]},"release_date":"2020-06-18T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["AppStream-8.3.0.GA:python38:3.8-8030020200818121840.4190259b"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2020:4641"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-06-18T00:00:00Z","ordinal":"5"}],"document_notes":{"note":["An update for the python38:3.8 module is now available for Red Hat Enterprise Linux 8.\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.","Python is an interpreted, interactive, object-oriented programming language, which includes modules, classes, exceptions, very high level dynamic data types and dynamic typing. Python supports interfaces to many system calls and libraries, as well as to various windowing systems.\nThe following packages have been upgraded to a later upstream version: python38 (3.8.3). (BZ#1847416)\nSecurity Fix(es):\n* PyYAML: command execution through python/object/apply constructor in FullLoader (CVE-2019-20477)\n* python: infinite loop in the tarfile module via crafted TAR archive (CVE-2019-20907)\n* PyYAML: arbitrary command execution through python/object/new when FullLoader is used (CVE-2020-1747)\n* python: wrong backtracking in urllib.request.AbstractBasicAuthHandler allows for a ReDoS (CVE-2020-8492)\n* python: DoS via inefficiency in IPv{4,6}Interface classes (CVE-2020-14422)\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\nAdditional Changes:\nFor detailed information on changes in this release, see the Red Hat Enterprise Linux 8.3 Release Notes linked from the References section.","This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original."]},"document_type":"Security Advisory"}}